dev-packages/application-package/src/application-props.ts

```diff
@@ -110,24 +106,34 @@ export interface FrontendApplicationConfig extends ApplicationConfig {
     /**
      * The default theme for the application. If not given, defaults to `dark`. If invalid theme is given, also defaults to `dark`.
      */
-    readonly defaultTheme: string;
+    readonly defaultTheme?: string;
 
     /**
      * The default icon theme for the application. If not given, defaults to `none`. If invalid theme is given, also defaults to `none`.
      */
-    readonly defaultIconTheme: string;
+    readonly defaultIconTheme?: string;
 
     /**
      * The name of the application. `Eclipse Theia` by default.
      */
-    readonly applicationName: string;
+    readonly applicationName?: string;
 
     /**
      * Electron specific configuration.
      */
     readonly electron?: Readonly<ElectronFrontendApplicationConfig>;
 }
 
+type PickRequired<T, K extends keyof T> = Omit<T, K> & Required<Pick<T, K>>;
+
+export namespace FrontendApplicationConfig {
+    export const DEFAULT: PickRequired<FrontendApplicationConfig, 'applicationName' | 'defaultTheme' | 'defaultIconTheme'> = {
+        applicationName: 'Eclipse Theia',
+        defaultTheme: 'dark',
+        defaultIconTheme: 'none'
+    };
+}
+
```

Comment:
Extending the design to other configurations makes sense however I'm not sure about predefined `required` and `partial` types, it seems like too much for me. What do you think?

Reply:
IIUC with your changes we already have two versions of the interfaces:

- `partial` is `FrontendApplicationConfig`.
- `resolved` is `FrontendApplicationConfig & typeof FrontendApplicationConfig.DEFAULT`.

Where `typeof FrontendApplicationConfig.DEFAULT` is a shortcut based on a runtime value that expands to `PickRequired<FrontendApplicationConfig, 'applicationName' | 'defaultTheme' | 'defaultIconTheme'>`.

So I thought it would be preferable to somehow get both interfaces as their own unique reference. Also note that on my branch, `FrontendApplicationConfig` still maps to an interface where fields with defaults are still required, only the `FrontendApplicationConfigProvider.set(...)` function now uses a more lax type.

Maybe you can just explicitly define both types so we can reference them without using `typeof SomeDefaultValue`, I think I'd be fine with just that. To define and export both `FrontendApplicationConfig` and `PartialFrontendApplicationConfig`.

Reply:
Reading again the code I realized that the only issue I had with your modifications was the naming haha. Making the first letter as uppercase would be nicer for a type.

TLDR; I'm ok with your modifications

Reply:
Please update your PR at will. My modifications were for illustration purpose, all that's required is to expose both interfaces separately, no matter how you prefer to do it :)

Reply:
I just understood why you named the properties starting with a lowercase. I don't know why but I though we could do `ConfigHelper<A, B>.Partial` haha. We can take your commit instead of mines, what do you think?

Reply:
https://github.com/eclipse-theia/theia/pull/9503

Reply:
Perfect, I'm closing the current PR :)